homeassistant/components/switch/rest.py

```diff
@@ -97,10 +112,22 @@ def turn_off(self, **kwargs):
 
     def update(self):
         """Get the latest data from REST API and update the state."""
-        request = requests.get(self._resource, timeout=10)
-        if request.text == self._body_on:
-            self._state = True
-        elif request.text == self._body_off:
-            self._state = False
+        request = requests.get(self._resource, timeout=self._timeout)
+
+        if self._value_template is not None:
+            response = template.render_with_possible_json_value(
+                self._hass, self._value_template, request.text, 'None')
+            response = response.lower()
+            if response == 'true':
```

Comment:
I agree with the value_template being renamed. And if this PR is accompanied with proper docs, it should be ok for the users.